Source/missiles.cpp

```diff
@@ -2193,19 +2171,23 @@ void AddFlash2(MissileStruct &missile, Point /*dst*/, int /*midir*/)
 
 void AddManashield(MissileStruct &missile, Point /*dst*/, int /*midir*/)
 {
-	auto &player = Players[missile._misource];
+	missile._miDelFlag = true;
 
-	if (player.pManaShield && !CanAddEffect(player, MIS_MANASHIELD)) {
-		missile._miDelFlag = true;
+	if (missile._misource < 0)
 		return;
-	}
 
-	missile._mirange = 48 * player._pLevel;
-	if (missile._micaster == TARGET_MONSTERS)
+	auto &player = Players[missile._misource];
+
+	if (missile._micaster == TARGET_MONSTERS) {
 		UseMana(missile._misource, SPL_MANASHIELD);
-	if (missile._misource == MyPlayerId)
+		player.pManaShield = !player.pManaShield;

```

Comment:
Not a fan of removing manshield, i don't see an upside and i could imagine mistake happening